common/src/test/java/com/graphicsfuzz/common/util/SideEffectCheckerTest.java

```diff
@@ -62,4 +65,12 @@ public void testCountingLoopHasSideEffects() throws Exception {
         ShadingLanguageVersion.ESSL_310));
   }
 
+  // TODO: Unignore when #521 is ready to merge.
+  @Ignore
+  @Test
```

Comment:
I would prefer this test if the expression was a more complete use of uaddCarry.

I suggest you write a simple shader with a main function that invokes uaddCarry in a few different ways, passing valid arguments each time and trying a few of the overloads of this function (if it has them; I think it does).

Then apply a CheckPredicateVisitor to the parsed translation unit for that function, and for each FunctionCallExpr visited (a) assert that the name of the called function is uaddCarry, and (b) that the expression is side-effect free.